ergo-wallet/src/main/scala/org/ergoplatform/wallet/AssetUtils.scala

```diff
@@ -7,22 +7,41 @@ import scala.collection.mutable
 object AssetUtils {
 
   @inline
-  def mergeAssetsMut(into: mutable.Map[ModifierId, Long], from: Map[ModifierId, Long]*): Unit = {
+  def mergeAssetsMut(into: mutable.Map[ModifierId, Long], from: TokensMap*): Unit = {
     from.foreach(_.foreach {
       case (id, amount) =>
-        into.put(id, into.getOrElse(id, 0L) + amount)
+        into.put(id, Math.addExact(into.getOrElse(id, 0L), amount))
     })
   }
 
   @inline
-  def mergeAssets(from: Map[ModifierId, Long], to: Map[ModifierId, Long] = Map.empty): Map[ModifierId, Long] = {
-    from.foldLeft(to) { case (acc, (id, amount)) =>
-      acc.updated(id, acc.getOrElse(id, 0L) + amount)
+  def mergeAssets(to: TokensMap, from: TokensMap*): TokensMap = {
+    from.foldLeft(to){ case (to, from) =>
+      from.foldLeft(to) { case (acc, (id, amount)) =>
+        acc.updated(id, Math.addExact(acc.getOrElse(id, 0L), amount))
+      }
+    }
+  }
+
+  @inline
+  def subtractAssets(from: TokensMap, subtractor: TokensMap*): TokensMap = {
+    subtractor.foldLeft(from){ case (from, substractor) =>
+      substractor.foldLeft(from) { case (acc, (id, amount)) =>
+        val currAmt = acc.getOrElse(id,
+          throw new IllegalArgumentException(s"Cannot subtract $amount of token $id: token not found in $acc"))
```

Comment:
Why checks are different from "subtractAssetsMut" method? Please unify.

Reply:
@kushti subtractAssetsMut is used in DefaultBoxSelector.formChangeBoxes, which in turn is used in applications. Adding this checks will make the method more strict. 
Can do it, as long as it is not a big deal.